tests/integration/components/self-clear-cache-test.js

```diff
@@ -0,0 +1,127 @@
+import { module, skip } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render, find } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+const EPOCH_TIMESTAMP = 1626242030;
+const HUMAN_TIME = ' July 13, 2021 10:53:50 PM IST';
+
+module('Integration | Component | self-clear-cache', (hooks) => {
+  setupRenderingTest(hooks);
+
+  skip('it shows last time user cleared the cache', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    await render(hbs`<SelfClearCache @lastClearedOn={{this.lastTime}} />`);
+
+    // Assert
+    assert.dom('[data-test-last-time]').hasText(HUMAN_TIME);
+  });
+
+  skip('it triggers cache clear on button click', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    const CACHE_CLEAR_CLICKED = 'CACHE_CLEAR_CLICKED';
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    this.set('onCacheClear', function () {
+      assert.step(CACHE_CLEAR_CLICKED);
+    });
+
+    await render(hbs`
+      <SelfClearCache
+        @lastClearedOn={{this.lastTime}}
+        @onCacheClear={{this.onCacheClear}}
+      />
+    `);
+
+    // Act
+    const btn = find('[data-test-btn-clear-cache]');
+    await click(btn);
+
+    // Assert
+    assert.verifySteps([CACHE_CLEAR_CLICKED]);
+  });
+
+  skip('it disables button after a click', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    const CACHE_CLEAR_CLICKED = 'CACHE_CLEAR_CLICKED';
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    this.set('onCacheClear', function () {
+      assert.step(CACHE_CLEAR_CLICKED);
+    });
+
+    await render(hbs`
+      <SelfClearCache
+        @lastClearedOn={{this.lastTime}}
+        @onCacheClear={{this.onCacheClear}}
+      />
+    `);
+
+    // Act
+    const btn = find('[data-test-btn-clear-cache]');
+    await click(btn);
+    await click(btn); // Another click -- This should not fire
```

Comment:
How are we knowing for which click action, the action is called?

Reply:
We set it up above as:

```
    this.set('onCacheClear', function () {
      assert.step(CACHE_CLEAR_CLICKED);
    });
```

Reply:
Then, in the act step, you can see that the test is expecting a button selector to exist (independent from class, tag name, id, or any other app specific code. Motivation here:
[Making your UI tests resilient to change](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change) and here [Ember Test Selectors](https://guides.emberjs.com/release/testing/#toc_ember-test-selectors)), and we click that button

```
// Act
const btn = find('[data-test-btn-clear-cache]');
```